x-pack/solutions/security/plugins/elastic_assistant/server/plugin.ts

```diff
@@ -40,6 +46,17 @@ import type { ConfigSchema } from './config_schema';
 import { attackDiscoveryAlertFieldMap } from './lib/attack_discovery/schedules/fields';
 import { ATTACK_DISCOVERY_ALERTS_CONTEXT } from './lib/attack_discovery/schedules/constants';
 
+interface FeatureFlagDefinition {
+  featureFlagName: string;
+  fallbackValue: boolean;
+  /**
+   * Function to execute when the feature flag is evaluated.
+   * @param enabled If the feature flag is enabled or not.
+   * @return `true` if susbscription needs to stay active, `false` if it can be unsubscribed.
+   */
+  fn: (enabled: boolean) => boolean | Promise<boolean>;
```

Comment:
The subscription/unsubscribe feels a bit complex and easy to get wrong, is it necessary or could we just keep the subscription always?